hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/cluster/strategy/PartitionAwareClusteringPlanStrategy.java

```diff
@@ -79,6 +79,11 @@ protected Stream<HoodieClusteringGroup> buildClusteringGroupsForPartition(String
         fileSliceGroups.add(Pair.of(currentGroup, numOutputGroups));
         currentGroup = new ArrayList<>();
         totalSizeSoFar = 0;
+
+        // if fileSliceGroups's size reach the max group, stop loop
+        if (fileSliceGroups.size() >= writeConfig.getClusteringMaxNumGroups()) {
```

Comment:
Looks reasonable, can we add a test in `TestSparkBuildClusteringGroupsForPartition` to test this early stopping improvement?

Reply:
okay

Reply:
test added. cc @stream2000